geo-traits/src/triangle.rs

```diff
@@ -115,3 +99,69 @@ impl<T> TriangleTrait for UnimplementedTriangle<T> {
         unimplemented!()
     }
 }
+
+impl<T> GeometryTrait for UnimplementedTriangle<T> {
+    type T = T;
+    type PointType<'b>
+        = UnimplementedPoint<Self::T>
+    where
+        Self: 'b;
+    type LineStringType<'b>
+        = UnimplementedLineString<Self::T>
+    where
+        Self: 'b;
+    type PolygonType<'b>
+        = UnimplementedPolygon<Self::T>
+    where
+        Self: 'b;
+    type MultiPointType<'b>
+        = UnimplementedMultiPoint<Self::T>
+    where
+        Self: 'b;
+    type MultiLineStringType<'b>
+        = UnimplementedMultiLineString<Self::T>
+    where
+        Self: 'b;
+    type MultiPolygonType<'b>
+        = UnimplementedMultiPolygon<Self::T>
+    where
+        Self: 'b;
+    type GeometryCollectionType<'b>
+        = UnimplementedGeometryCollection<Self::T>
+    where
+        Self: 'b;
+    type RectType<'b>
+        = UnimplementedRect<Self::T>
+    where
+        Self: 'b;
+    type TriangleType<'b>
+        = UnimplementedTriangle<Self::T>
+    where
+        Self: 'b;
+    type LineType<'b>
+        = UnimplementedLine<Self::T>
+    where
+        Self: 'b;
+
+    fn dim(&self) -> Dimensions {
+        unimplemented!()
+    }
+
+    fn as_type(
+        &self,
+    ) -> GeometryType<
+        '_,
+        Self::PointType<'_>,
+        Self::LineStringType<'_>,
+        Self::PolygonType<'_>,
+        Self::MultiPointType<'_>,
+        Self::MultiLineStringType<'_>,
+        Self::MultiPolygonType<'_>,
+        Self::GeometryCollectionType<'_>,
+        Self::RectType<'_>,
+        Self::TriangleType<'_>,
+        Self::LineType<'_>,
+    > {
+        GeometryType::Triangle(self)
+    }
+}
```

Comment:
We should be able to put this in a macro similar to https://github.com/georust/geo/blob/750ed14a2f01c6a047e2ced76ab12261ef46ba5a/geo-traits/src/geometry.rs#L291-L438

Maybe in that same file?
https://github.com/georust/geo/blob/750ed14a2f01c6a047e2ced76ab12261ef46ba5a/geo-traits/src/geometry.rs#L440-L447

Reply:
Pushed this change to this branch in [`f834667` (#1346)](https://github.com/georust/geo/pull/1346/commits/f8346671ca3c37b2534f65010f7e9b4b5a77256f), and then realized maybe I should double check that's ok etiquette? Not sure when it's frowned upon to push to another's branch. It did minimize the diff by a lot, but maybe I should've created another PR into this PR?

Reply:
I wouldn't say it's a question of etiquette, but a practical issue: it can surprisingly change the source branch (until you run `log` or whatever). A PR to the source branch is probably better (even though it always feels more finicky to me) because then the decision about what to do with the change is in the hands of the original author; they can pull it, comment on it, ask for changes themselves etc.

Reply:
That makes a lot of sense; sorry if I inconvenienced you @frewsxcv !

Reply:
I do prefer getting consent from the PR author before pushing to their branches, but in this case I really didn't mind so thanks for the commit! I was wondering if a macro would've been more appropriate for that